string filename
---
int32 success